lang/cs/Org.Apache.REEF.Tang.Tests/Injection/TestInjection.cs

```diff
@@ -304,6 +304,55 @@ public void TestExternalObject()
 
             Assert.IsNotNull(o.ExternalObject is ExternalClass);
         }
+        
+        /// <summary>
+        /// In this test, interface is generic of T. Implementations use different T as type. 
+        /// When doing injection, we need to get type of generic type T so that to get generic interface
+        /// </summary>
+        [TestMethod]
+        public void TestInjectionWithT()
```

Comment:
This test is not for a new feature in Tang, right? Still, it seems like a good candidate for a separate pull request.